lib/connectors/memory.js

```diff
@@ -252,6 +252,36 @@ Memory.prototype.updateOrCreate = function(model, data, options, callback) {
   });
 };
 
+Memory.prototype.patchOrCreateWithWhere =
+Memory.prototype.upsertWithWhere = function(model, where, data, options, callback) {
+  var self = this;
+  var primaryKey = this.idName(model);
+  var filter = { where: where };
+  var nodes = self._findAllSkippingIncludes(model, filter);
+  if (nodes.length === 0) {
+    return self._createSync(model, data, function(err, id) {
+      if (err) return process.nextTick(function() { callback(err); });
+      self.saveToFile(id, function(err, id) {
+        self.setIdValue(model, data, id);
+        callback(err, self.fromDb(model, data), { isNewInstance: true });
+      });
+    });
+  }
+  if (nodes.length === 1) {
```

Comment:
as a nit though, we should probably use else if since you use if/else right after anyways